### DISCUSSION TITLE
Atualização de valor selecionado po-combo

### DISCUSSION THREAD

**Discussion by danielscheeren** [po-ui/po-angular#1935] [Angular]
Boa tarde.

Estou implementando um po-combo e realizando algumas tratativas. Uma delas seria adicionar uma das opções como o valor incial (selecionado).
Tentei utilizar o ngmodel para a atualização, porém não surte efeito. O mesmo também não permite fazer uso do two-way binding informando o erro: **The property and event halves of the two-way binding 'ngModel' are not bound to the same target.**.

Estou me baseando no exemplo da documentação:
![image](https://github.com/po-ui/po-angular/assets/45018228/33c319a6-6413-4022-b741-e577aaeb8ab8)

Ao utilizar o [ngModel] para setar a informação, nada acontece:
![image](https://github.com/po-ui/po-angular/assets/45018228/b989c2cd-fdf0-4f13-ad6f-c1c1166c74d0)

**produtos-list.component.html**
` 
<po-page-default p-title="Produtos" [p-actions]="actions" [p-breadcrumb]="breadcrumb">
    
    <po-widget class="po-lg-12">
        <form #fOption="ngForm">
            
            <po-combo ngDefaultControl
                name="combo"
                p-label="Selecione:"
                [ngModel]="selectedOption"
                (p-change)="seleciona_opcao($event)"
                [p-options]="options"
                bindLabel="label">
            </po-combo>

            <po-divider
                p-border-width="string" >
            </po-divider>
            
            <po-table
                [p-hide-text-overflow]="true"
                [p-show-more-disabled]="disableNext"
                p-container [p-sort]="true"
                [p-striped]="true"
                [p-columns]="columns"
                [p-items]="items"
                (p-show-more)="getItems(true)">
            </po-table>
            
        </form>
    </po-widget>

</po-page-default>
`

**produtos-list.component.ts**
`
 import { Component, EventEmitter, OnInit, Output, ViewChild } from '@angular/core';
// import { ClienteListService } from './produtos-list.service';
import { Router } from '@angular/router'
import { PoPageAction, PoBreadcrumb, PoTableColumn, PoDialogService, PoNotificationService, PoComboOption, PoComboOptionGroup } from '@po-ui/ng-components';
import { lastValueFrom } from 'rxjs'
import { NgForm } from '@angular/forms';

// services
import { ProdutosListService } from './produtos-list.service';

@Component({
  selector: 'app-produtos-list',
  templateUrl: './produtos-list.component.html',
  styleUrls: ['./produtos-list.component.css']
})
export class ProdutosListComponent implements OnInit {
  @ViewChild('fOption', { static: true }) form: NgForm;

  items: Array<any> = [];
  page = 1;
  disableNext = false;
  selectedOption: string = 'Teste1';

  public readonly actions: Array<PoPageAction> = [
    { label: 'Atualizar', action: this.getItems.bind(this), icon: 'po-icon-refresh' },
    { label: 'Novo', action: () => { this.router.navigate(['/client-edit']) }, icon: 'po-icon-plus' },
  ];

  public readonly breadcrumb: PoBreadcrumb = {
    items: [{ label: 'Home', link: '/' }, { label: 'Produtos' }]
  };
  
  public readonly columns: Array<PoTableColumn> = [
    { property: 'b1_cod', width: '10%', label: 'Código' },
    { property: 'b1_desc', width: '30%', label: 'Descrição' },
    { property: 'b1_grupo', width: '40%', label: 'Grupo' },
    { property: 'bm_desc', width: '40%', label: 'Descrição Grupo' }
  ]

  public readonly options: Array<PoComboOption | PoComboOptionGroup> = [   // para criar agrupadores Array<PoComboOption | PoComboOptionGroup>
    { value: '1', label: 'Teste1'},
    { value: '2', label: 'Teste2'},
    { value: '3', label: 'Teste3'},
    { value: '4', label: 'Teste4'}
  ];

  constructor(
    private produtosListService: ProdutosListService,
    private poDialog: PoDialogService,
    public router: Router,
    private poNotification: PoNotificationService,
  ) { }

  ngOnInit() {
    this.getItems();
  }

  getItems(lShowMore = false) {

    if (lShowMore) {
      this.page++
    } else {
      this.items = [{ }]
    }

    this.produtosListService
      .get(this.page).subscribe(
        {
          next: (data: any) => {
            // console.log(`get ${this.page}`)
            // console.log("data", data)
            this.items = [...data.items];
            this.disableNext = data.disablenext;
          }

        }
      )

  }

  
  @Output() openMultiselectEventHandler: EventEmitter<any> = new EventEmitter();
  @Output() closeMultiselectEventHandler: EventEmitter<any> = new EventEmitter();

  // evento ao selecionar opção do combo
  seleciona_opcao(value: any) {
    
    if (value == '1') {
      this.getItems()
    } else if (value == '2') {
      this.getItems()
    } else if (value == '3') {
      this.getItems()
    } else if (value == '4') {
      this.getItems()
    }

    this.closeMultiselectEventHandler.emit(null)
  }
  
}

`

Versão que estou utilizando:
![image](https://github.com/po-ui/po-angular/assets/45018228/33be9a20-ca58-41b8-a634-de5a179e8e42)

**Comment by anliben** [comment]
Boa Tarde @danielscheeren, tudo bem?

Iremos analisar e em breve iremos responder sua solicitação.

attr.

**Reply by kleberrogersantos** [reply]
Boa tarde. 
Tudo bem?
Tenho a mesma situação, há alguma solução ?

Obrigado

**Reply by leandroayala** [reply]
Estou com esse mesmo problema.

**Comment by rebecaas** [comment]
Encontrei esse mesmo problema usando o angular  19.1.0. Alguém conseguiu solucionar?

**Comment by alinelariguet** [comment]
Bom dia pessoal, na doc do combo fala sobre o ngModel, por favor verifiquem se isso resolve o problema reportado:
![image](https://github.com/user-attachments/assets/2bc9b9b2-2756-4024-9db6-a4380e09b9d3)
Att.
